### PR TITLE
Add supplementing annotations to canvas

### PR DESCRIPTION
**Pull request by masaball**
This PR adds builders for the V3 manifest to facilitate creating annotations for supplementing content, most specifically [A/V caption/subtitle and transcript content](https://iiif.io/api/cookbook/recipe/0219-using-caption-file/).

### CODE REVIEW

File: lib/iiif_manifest/v3/manifest_service_locator.rb [hunk 1]
```diff
@@ -88,6 +89,14 @@ def choice_builder
           )
         end
 
+        def supplementing_content_builder
+          IIIFManifest::ManifestServiceLocator::InjectedFactory.new(
+            ManifestBuilder::SupplementingContentBuilder,
+            iiif_annotation_factory: iiif_annotation_factory,
+            body_builder_factory: body_builder_factory
```
Comment (cjcolvar) [inline lib/iiif_manifest/v3/manifest_service_locator.rb:96]: I think this needs to be the `supplementing_body_builder_factory` in order for the language property to be added.
```suggestion
            body_builder_factory: supplementing_body_builder_factory
```

**Review by cjcolvar — COMMENTED** [review]
This looks pretty good.  I think there a couple small issues to fix.  One of them got me thinking:  would it work to add support for `language` to `BodyBuilder` and skip adding a new body builder for supplementing content?  `BodyBuilder` currently assumes it will be used for `painting` annotations in the `items` of a canvas so reusing it for supplementing annotations might be a bit odd but they are doing the same thing, creating the body of an annotation.  Similarly `SupplementingContent` is only a small change from `DisplayContent`.  What do you think about reusing those and extending them to support this case?  Is that too confusing?  Is making separate classes a lot cleaner and worth the duplication?

File: lib/iiif_manifest/v3/manifest_builder/supplementing_body_builder.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        end
+
+        private
+
+        def build_supplementing_body
+          supplementing_body['id'] = supplementing_content.url
+          supplementing_body['type'] = supplementing_content.try(:type) || 'Text'
+          supplementing_body['format'] = supplementing_content.format if supplementing_content.try(:format).present?
+          supplementing_body['label'] = supplement_label
+          supplementing_body['language'] = supplementing_content.try(:language) || 'eng'
```
[the remark this answers is not in the document]
Comment (mathewjordan) [reply lib/iiif_manifest/v3/manifest_builder/supplementing_body_builder.rb:23]: Could default be `none` in this instance?

[the remark this answers is not in the document]
Comment (cjcolvar) [reply lib/iiif_manifest/v3/manifest_builder/supplementing_body_builder.rb:23]: :+1: That would be consistent with how `iiif_manifest` handles language maps.

**Review by cjcolvar — APPROVED** [review]
Looks good!

**Pull request merged**